src/Entity/RequestRepository.php

```diff
@@ -46,6 +63,27 @@ public function getRequestCountByIp(string $ip): int
             $requestItem->addRequest();
             $em->persist($requestItem);
 
+            try {
+                $em->flush();
+            } catch (UniqueConstraintViolationException $exception) {
+                $logLevel = $previousTries === 0 ? LogLevel::WARNING : LogLevel::ERROR;
+                $this->logger->log(
+                    $logLevel,
+                    sprintf(
+                        'UniqueConstraintViolationException led to access insert retry for IP %s',
+                        $ip
+                    )
+                );
+
+                if ($previousTries > 0) {
+                    // "Fail safe" by simulating a very high request count if something is going persistently wrong.
+                    return 100000;
```

Comment:
This doesn't appear to match the expected return type. On line 87 it appears to return a `$requestItem`, not an int. But maybe I'm missing something

Reply:
Oops, good catch @rasmuswinter. Should be fixed now.